LocalyticsPlugin/Assets/Localytics/Localytics.cs

```diff
@@ -8,1073 +8,2163 @@
+
+		public static bool OptedOut
+		{
+			get
+			{
+				#if UNITY_ANDROID
+				return LocalyticsClass.CallStatic<bool>("isOptedOut");
+				#elif UNITY_IOS
+				return _isOptedOut();
+				#else
+				throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+				#endif
+			}
+			set
+			{
+				#if UNITY_ANDROID
+				LocalyticsClass.CallStatic("setOptedOut", value);
+				#elif UNITY_IOS
+				_setOptedOut(value);
+				#else
+				throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+				#endif
+			}
+		}
+
+		public static void RegisterForAnalyticsEvents ()
 		{
 			#if UNITY_ANDROID
-			LocalyticsClass.CallStatic("setProfileAttribute", attributeName, attributeValue, GetProfileScopeEnum(scope));
+			if (_analyticsListener == null)
+			{
+				_analyticsListener = new AnalyticsListener();
+			}
+
+			LocalyticsClass.CallStatic("setAnalyticsListener", _analyticsListener);
 			#elif UNITY_IOS
-			string values = MiniJSON.jsonEncode (new string[] { attributeValue });
-			_removeProfileAttributesFromSet (values, attributeName, (int)scope);
+			_registerReceiveAnalyticsCallback (ReceiveAnalyticsMessage);
 			#else
-			throw new NotImplementedException("Localytics Unity SDK only supports iOS or Android");
+			throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
 			#endif
 		}
-		
-		public static void SetProfileAttribute(string attributeName, string[] attributeValue, ProfileScope scope = ProfileScope.Application)
+
+		public static void UnregisterForAnalyticsEvents ()
 		{
 			#if UNITY_ANDROID
-			LocalyticsClass.CallStatic("setProfileAttribute", attributeName, attributeValue, GetProfileScopeEnum(scope));
+			if (_analyticsListener == null)
+			{
+				return;
+			}
+
+			LocalyticsClass.CallStatic("setAnalyticsListener", null);
 			#elif UNITY_IOS
-			string values = MiniJSON.jsonEncode (attributeValue);
-			_removeProfileAttributesFromSet (values, attributeName, (int)scope);
+			_removeAnalyticsCallback ();
 			#else
-			throw new NotImplementedException("Localytics Unity SDK only supports iOS or Android");
+			throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
 			#endif
 		}
-		
-		public static void TagEvent(string eventName, Dictionary<string, string> attributes = null, long customerValueIncrease = 0)
+
+		// #################################################
+		// Profiles
+		// #################################################
+
+		public static void SetProfileAttribute (string attributeName, long attributeValue, ProfileScope scope = ProfileScope.Application)
 		{
 			#if UNITY_ANDROID
-			LocalyticsClass.CallStatic("tagEvent", eventName, DictionaryToMap(attributes), customerValueIncrease);
+			LocalyticsClass.CallStatic("setProfileAttribute", attributeName, attributeValue, GetProfileScopeEnum(scope));
```

Comment:
On Android the standard events methods take object types instead of primitives (to allow for `null`):
```
public static void tagPurchased(@Nullable final String itemName, @Nullable final String itemId, @Nullable final String itemType, @Nullable final Long itemPrice, @Nullable final Map<String, String> attributes);
```

From what I've seen in C# there isn't an object version of primitives (e.g. `long` -> `Long` or `int` -> `Integer`), so we can't support a `null` value. However, we still need to wrap it in the `Object` version when using `CallStatic` otherwise the bridging code won't find the correct method signature.